wgsl/index.bs

```diff
@@ -2534,19 +2534,31 @@ declaration of the identifier as a type alias or structure type.
   </xmp>
 </div>
 
-# `var` and `let` # {#variables}
+# `var` and `let` # {#var-and-let}
 
-TODO: *Stub* (describe what a constant is): A constant is a name for a value, declared via a `let` declaration.
-What types are permitted?  Storable, plus pointer to store type.
+A <dfn noexport>let declaration</dfn> specifies a name for a value.
+Once the value for a let-declaration is computed, it is immutable.
+When an identifier use [=resolves=] to a let-declaration, the identifier denotes that value.
 
-TODO(dneto): A `let` may not be of type pointer-to-handle. A function parameter may not have type pointer-to-handle.
-Otherwise we'd have a need to make a pointer-to-handle type expression. But we've reserved the [=storage classes/handle=] keyword.
-When translating from SPIR-V, you must trace through
-the OpCopyObject (or no-index OpAccessChain) instructions that might be between
-the pointer-to-array and the pointer-to-struct.
+When a `let` identifier is declared without an explicitly specified type,
+e.g. `let foo = 4`, the type is automatically inferred from the expression to the right of the equals token (`=`).
+If the type is specified, e.g `let foo: i32 = 4`, it must exactly match the type of the initializer expression.
```

Comment:
should the load rule be mentioned here? AFAIK, it contradicts the "exactly match" part

Reply:
Indeed.  The subtle but proper way to phrase is to say that the value must evaluate to the declared type.
Then the load rule takes effect automatically. I'll patch it to improve this.